docs/notebooks/batch_optimization.pct.py

```diff
@@ -0,0 +1,176 @@
+#
+# Note that both of these acquisition functions have controllable parameters. In particular, `BatchMonteCarloExpectedImprovement` is computed using a Monte-Carlo method (so it requires a `sample_size`), but uses a reparametrisation trick to make it deterministic. `LocallyPenalizedExpectedImprovement` has parameters controlling the degree of penalization that must be estimated from a random sample of  `num_samples` model predictions.
+#
+# We now visualize the batch of 10 points chosen by each of these methods overlayed on the standard `ExpectedImprovement` acquisition function. `BatchMonteCarloExpectedImprovement` chooses a more diverse set of points, whereas `LocallyPenalizedExpectedImprovement` focuses evaluations into more promising areas of the space.
+
+# %%
+from trieste.acquisition import (
+    BatchMonteCarloExpectedImprovement,
+    ExpectedImprovement,
+    LocallyPenalizedExpectedImprovement,
+)
+from trieste.acquisition.rule import EfficientGlobalOptimization
+
+ei = ExpectedImprovement()
+ei_acq_function = ei.using(OBJECTIVE).prepare_acquisition_function(initial_data, models)
+plot_acq_function_2d(ei_acq_function, [0, 0], [1, 1], contour=True, grid_density=100)
+
+batch_ei = BatchMonteCarloExpectedImprovement(sample_size=1000)
+chosen_points, _ = EfficientGlobalOptimization(
+    num_query_points=10, builder=batch_ei.using(OBJECTIVE)
+).acquire(search_space, initial_data, models)
+plt.scatter(
+    chosen_points[:, 0],
+    chosen_points[:, 1],
+    color="red",
+    lw=5,
+    label="Batch-EI",
+    marker="*",
+    zorder=1,
+)
+
+local_penalization = LocallyPenalizedExpectedImprovement(search_space, num_samples=1000)
+chosen_points, _ = EfficientGlobalOptimization(
+    num_query_points=10, builder=local_penalization.using(OBJECTIVE)
+).acquire(search_space, initial_data, models)
+plt.scatter(
+    chosen_points[:, 0],
+    chosen_points[:, 1],
+    color="cyan",
+    lw=10,
+    label="Local \nPenalization",
+    marker="+",
+)
+
+plt.legend(bbox_to_anchor=(1.2, 1), loc="upper left")
+plt.xlabel(r"$x_1$")
+plt.ylabel(r"$x_2$")
+cbar = plt.colorbar()
+cbar.set_label("EI", rotation=270)
+
+# %% [markdown]
+# ## Run the batch optimization loop
+# We can now run a batch Bayesian optimization loop by defining a `BayesianOptimizer` with one of our batch acquisition functions.
+#
+# We'll run each method for ten steps for batches of three points.
+
+# %%
+from trieste.acquisition.rule import EfficientGlobalOptimization
+
+bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
+
+batch_ei_rule: EfficientGlobalOptimization[Box] = EfficientGlobalOptimization(
```

Comment:
I find it confusing that we define the rule twice in this notebook, in different ways.

Reply:
also, remove `[Box]` stuff?